app.rb

```diff
@@ -0,0 +1,98 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'Empty list'
+    else
+      @books.each do |book|
+        puts "Title: #{book.title} Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'Empty list'
+    else
+      @people.each do |person|
+        if person.is_a?(Student)
+          puts "[Student] Person created: ID: #{person.id}, Name: #{person.name}, Age: #{person.age}, " \
+               "Permission: #{person.parent_permission}"
+
+        elsif person.is_a?(Teacher)
+          puts "[Teacher] Person created: ID: #{person.id}, Name: #{person.name}, Age: #{person.age}, " \
+               "specialization: #{person.specialization}"
+        end
+      end
+    end
+  end
+
+  def create_person(name, age, type, extra_info)
+    if type == 'student'
+      person = Student.new(name, age, parent_permission: extra_info[:parent_permission])
+    elsif type == 'teacher'
+      person = Teacher.new(name, age, extra_info[:specialization])
+    else
+      puts 'Invalid person type!'
+      return
+    end
+
+    @people.push(person)
+    puts "Person created: ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book created: Title: #{book.title}, Author: #{book.author}"
+  end
+
+  def get_book_by_index(book_index)
+    @books[book_index] if book_index >= 0 && book_index < @books.length
+  end
+
+  def get_person_by_index(person_index)
+    @people[person_index] if person_index >= 0 && person_index < @people.length
+  end
+
+  def create_rental(book, person, date)
+    if book && person
+      rental = Rental.new(date, book, person)
```

Comment:
- [ ] `required`
- [ ] Good job so far! 👏 
- [ ] However everything is working okay until i try creating rental date i'm getting an error 😞 
- [ ] Please see the screenshot below

![image](https://github.com/Metaverse-Nabeel/OOP-school-library/assets/66230306/cedc9eae-c706-4bb2-bd13-eec237186bce)

Reply:
Hello @mwanawabangona

I have made the changes and now it is working perfectly fine. Please review it again. I have submitted the request for review.